src/analysis/normed_space/conformal.lean

```diff
@@ -0,0 +1,211 @@
+def conformal_at {X Y : Type*} [inner_product_space ℝ X] [inner_product_space ℝ Y]
+  (f : X → Y) (x : X) :=
+∃ (f' : X →L[ℝ] Y), has_fderiv_at f f' x ∧ is_conformal_map f'
+
+
+namespace conformal_at
+
+variables {X Y Z : Type*}
+  [inner_product_space ℝ X] [inner_product_space ℝ Y] [inner_product_space ℝ Z]
+
+open linear_isometry_equiv continuous_linear_map
+
+theorem differentiable_at {f : X → Y} {x : X} (h : conformal_at f x) :
+  differentiable_at ℝ f x :=
+let ⟨_, h₁, _, _, _, _⟩ := h in h₁.differentiable_at
+
+theorem id (x : X) : conformal_at id x :=
+⟨id ℝ X, has_fderiv_at_id _, 1, one_ne_zero, refl ℝ X, by ext; simp⟩
+
+theorem const_smul {c : ℝ} (h : c ≠ 0) (x : X) : conformal_at (λ (x': X), c • x') x :=
+⟨c • continuous_linear_map.id ℝ X, has_fderiv_at.const_smul (has_fderiv_at_id x) c,
+  c, h, refl ℝ X, by ext; simp⟩
+
+theorem comp {f : X → Y} {g : Y → Z} {x : X} :
+  conformal_at f x → conformal_at g (f x) → conformal_at (g ∘ f) x :=
+begin
+  rintro ⟨f', hf₁, cf, hcf, lief, hf₂⟩ ⟨g', hg₁, cg, hcg, lieg, hg₂⟩,
+  exact ⟨g'.comp f', has_fderiv_at.comp x hg₁ hf₁, cg * cf, mul_ne_zero hcg hcf, lief.trans lieg,
+    by { ext, simp [coe_comp' f' g', hf₂, hg₂, smul_smul cg cf] }⟩,
+end
+
+/-- A real differentiable map `f` is conformal at point `x` if and only if
+    its differential `f'` at that point scales any inner product by a positive scalar. -/
+theorem _root_.conformal_at_iff {f : X → Y} {x : X} {f' : X ≃L[ℝ] Y}
+  (h : has_fderiv_at f f'.to_continuous_linear_map x) :
+  conformal_at f x ↔ ∃ (c : ℝ) (hc : 0 < c),
+  ∀ (u v : X), inner (f' u) (f' v) = (c : ℝ) * (inner u v) :=
+begin
+  split,
+  { rintros ⟨f₁, h₁, c₁, hc₁, lie, h₂⟩,
+    refine ⟨c₁ * c₁, mul_self_pos hc₁, λ u v, _⟩,
+    simp [← f'.coe_coe, ← f'.coe_def_rev, has_fderiv_at.unique h h₁, h₂, inner_map_map,
+          real_inner_smul_left, real_inner_smul_right, mul_assoc] },
+  { rintro ⟨c₁, hc₁, huv⟩,
+    let c := real.sqrt c₁⁻¹,
+    have hc : c ≠ 0 := λ w, by {simp only [c] at w;
+      exact (real.sqrt_ne_zero'.mpr $ inv_pos.mpr hc₁) w},
+    let c_map := linear_equiv.smul_of_ne_zero ℝ Y c hc,
+    let f₁ := f'.to_linear_equiv.trans c_map,
+    have minor : (f₁ : X → Y) = c • f' := rfl,
+    have minor' : (f' : X → Y) = c⁻¹ • f₁ := by ext;
+      simp_rw [minor, pi.smul_apply]; rw [smul_smul, inv_mul_cancel hc, one_smul],
+    have key : ∀ (u v : X), inner (f₁ u) (f₁ v) = inner u v := λ u v, by
+      simp_rw [minor, pi.smul_apply]; rw [real_inner_smul_left,
+          real_inner_smul_right, huv u v, ← mul_assoc, ← mul_assoc,
+          real.mul_self_sqrt $ le_of_lt $ inv_pos.mpr hc₁,
+          inv_mul_cancel $ ne_of_gt hc₁, one_mul],
+    exact ⟨f'.to_continuous_linear_map, h, c⁻¹, inv_ne_zero hc,
+           f₁.isometry_of_inner key, minor'⟩, },
+end
+
+/-- The characteristic function of a conformal map. -/
+def char_fun_at {f : X → Y} (x : X) {f' : X ≃L[ℝ] Y}
+  (h : has_fderiv_at f f'.to_continuous_linear_map x) (H : conformal_at f x) : ℝ :=
```

Comment:
Did you find a reference calling this the "characteristic function"?  I would tend to call this the "conformal factor".

Reply:
Thanks for the comments. I did see the name conformal factors but I was not sure in what context they are for. I could change the name.

Reply:
If you have also seen the name "characteristic function" used in the literature, I think it's ok to use that name (although I would include a mention of the alternative name in the docstring).  I just wanted to make sure it wasn't a name you invented yourself.